tests/facet_search_settings.test.ts

```diff
@@ -0,0 +1,176 @@
+import { afterAll, expect, test, describe, beforeEach } from "vitest";
+import { ErrorStatusCode } from "../src/types";
+import {
+  clearAllIndexes,
+  config,
+  BAD_HOST,
+  MeiliSearch,
+  getClient,
+  dataset,
+} from "./utils/meilisearch-test-utils";
+
+const index = {
+  uid: "movies_test",
+};
+
+afterAll(() => {
+  return clearAllIndexes(config);
+});
+
+describe.each([{ permission: "Master" }, { permission: "Admin" }])(
+  "Test on facet search settings",
+  ({ permission }) => {
+    beforeEach(async () => {
+      await clearAllIndexes(config);
+      const client = await getClient("Master");
+      const { taskUid } = await client.index(index.uid).addDocuments(dataset);
+      await client.waitForTask(taskUid);
+    });
+
+    test(`${permission} key: Get facetSearch settings on empty index`, async () => {
+      const client = await getClient(permission);
+
+      const response = await client.index(index.uid).getFacetSearch();
+      expect(response).toMatchSnapshot();
+    });
+
+    test(`${permission} key: Set facetSearch settings with dedicated endpoint on empty index`, async () => {
+      const client = await getClient(permission);
+
+      const { taskUid } = await client
+        .index(index.uid)
+        .updateFacetSearch(false);
+      await client.index(index.uid).waitForTask(taskUid);
+
+      const updatedSettings = await client.index(index.uid).getFacetSearch();
+      expect(updatedSettings).toBe(false);
```

Comment:
Why do you `expect(response).toMatchSnapshot();` for the other tests but not this one? 👀

Reply:
I tried to implement the tests following the same logic as other tests. Maybe I got this one wrong.

My understanding is that the tests with `expect(response).toMatchSnapshot();` are testing that they receive the "default" settings object. They're not testing for specific field value.

Here, I'm testing that the `updateFacetSearch` method behaves correctly: so I'm checking for a specific value returned by Meilisearch. I could match it against the entire snapshot, too, but it seemed out of scope for this specific test.

Does my reasoning make sense?

Reply:
Thank you for your answer 🙏 
It does make sense indeed 👍  Let's keep it like this then!